addon/-private/serializers/embedded-records-mixin.js

```diff
@@ -321,16 +372,38 @@ export default Ember.Mixin.create({
       this._super(snapshot, json, relationship);
       return;
     }
-    var includeIds = this.hasSerializeIdsOption(attr);
-    var includeRecords = this.hasSerializeRecordsOption(attr);
-    if (includeIds) {
+
+    if (this.hasSerializeIdsOption(attr)) {
       let serializedKey = this.keyForRelationship(attr, relationship.kind, 'serialize');
       json[serializedKey] = snapshot.hasMany(attr, { ids: true });
-    } else if (includeRecords) {
+    } else if (this.hasSerializeRecordsOption(attr)) {
       this._serializeEmbeddedHasMany(snapshot, json, relationship);
+    } else if (isEnabled("ds-serialize-ids-and-types") && this.hasSerializeIdsAndTypesOption(attr)) {
```

Comment:
If I am not mistaken this is not yet possible with feature flags (aka the code will not be stripped correctly when the feature flag is disabled).

Unfortunately you need to split this up:

``` js
} else {
  if (isEnabled("ds-serialize-ids-and-types")) {
    if (this.hasSerializeIdsAndTypesOption(attr)) {
      this._serializeHasManyAsIdsAndTypes(snapshot, json, relationship);
    }
  }
}
```